resources/splunk_install.rb

```diff
@@ -0,0 +1,219 @@
+# Encoding: utf-8
+
+class SplunkInstall < ChefCompat::Resource
+  include PlatformHelpers, PathHelpers
+
+  resource_name :splunk_install
+
+  property :name, String, name_property: true, identity: true
+  property :package, [:splunk, :universal_forwarder], required: true
+  property :service_user, String, default: 'splunk'
```

Comment:
Is there a use for doing so? Having a system user assumes there's a group named after the user, so that's not going to change. We could allow adding extra groups, but consumers could also do that. It might avoid cloning the user resource, however.